### PR TITLE
Fix stucking wait mode caused by unscheduled workloads and disconnected agents variations

### PR DESCRIPTION
**Pull request by inf17101**
Issues: #320 

The PR covers the following:
- fix the stucking wait mode when pending initial workloads with disconnected agents are deleted
- display the deleted pending initial workloads as removed properly in the wait list
- fix the stucking wait mode when workloads are updated to be unscheduled (empty agent name) and vice versa
- fix the stucking wait mode when unscheduled workloads are deleted and display them properly as removed in the wait list
- fixing the Ankaios server sending a removed execution state when a pending initial workload is deleted and its agent is disconnected
- fix the stucking wait mode when a workload is created and during the creation the agent has been disconnected
- fix the stucking wait mode when a workload is scheduled for deletion on the runtime and the agent is disconnected during the delete operation
- fix a gap in the `ank get workload` swdd requirements by creating a new swdd for not described functionality when constructing the list of workloads
- creating new swdds or changing existing swdds: here I have tried to keep the amount of requirements in the cli at a minimum, can be discussed in the review if it is enough or if we need to split them a little bit more.

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

**Comment by inf17101** [comment]
@krucod3: Do you think we need to change precondition for the following swdds after fixing the gap in `ank get workloads`?:

- `swdd~cli-shall-sort-list-of-workloads~1`
- `swdd~cli-shall-filter-list-of-workloads~1`
- `swdd~cli-shall-present-list-of-workloads~1`

to point out more that sorting, filtering and so on is done after the CLI has created the initial list of workloads with their runtime names based on the current CompleteState? This is then closer to the code.

**Review by krucod3 — COMMENTED** [review]
Review done.

All looks good, just had two small findings regarding the requirements. 

I'll test a bit manually and can merge afterwards when the small things are fixed.

File: ank/src/cli_commands/get_workloads.rs [hunk 1]
```diff
@@ -123,6 +129,7 @@ mod tests {
     }
 
     // [utest->swdd~cli-provides-list-of-workloads~1]
+    // [utest->swdd~processes-complete-state-to-list-workloads~1]
```
[the remark this answers is not in the document]
Comment (krucod3) [reply ank/src/cli_commands/get_workloads.rs:132]: We can resolve this directly.

**Comment by inf17101** [comment]
> Review done.
> 
> All looks good, just had two small findings regarding the requirements.
> 
> I'll test a bit manually and can merge afterwards when the small things are fixed.

@krucod3: I have fixed the findings. There is one comment left about the preconditions of three requirements. Can you check this as well?

**Review by krucod3 — APPROVED** [review]
LGTM 👍

**Pull request merged**